aptos-move/framework/aptos-framework/sources/configs/Stake.move

```diff
@@ -908,8 +913,20 @@ module AptosFramework::Stake {
             remaining_lockup_time = stake_pool.locked_until_secs - current_time;
         };
 
-        let rewards_amount = distribute_reward(&mut stake_pool.active, num_blocks, num_successful_votes, remaining_lockup_time, validator_set_config);
-        rewards_amount = rewards_amount + distribute_reward(&mut stake_pool.pending_inactive, num_blocks, num_successful_votes, remaining_lockup_time, validator_set_config);
+        // Merge rewards into the active stake if the validator still has remaining active stake.
+        let rewards = mint_reward(
+            validator.voting_power,
+            num_blocks,
+            num_successful_votes,
+            remaining_lockup_time,
+            validator_set_config,
+        );
+        let rewards_amount = Coin::value<TestCoin>(&rewards);
+        if (Coin::value<TestCoin>(&stake_pool.active) > 0) {
```

Comment:
why we care about if active is 0? if it's 0 it'll be kicked out from validator anymore (assume minimum_stake > 0)

Reply:
Because here the validator might not have been removed from the active set - reward distribution happens before the new validator set is computed. The idea is that, if they're still active and have non-zero active stake, rewards will be added to the active pool. Otherwise, it'd get added to inactive (indirectly via pending_inactive).

Reply:
what's the problem of always adding them to active?

Reply:
Updated